### ISSUE TITLE
Setting AnchorPreset in _Ready has no effect

### ISSUE THREAD

**Issue by dotlogix** [godotengine/godot#67161]
### Godot version

4.0.dev (17c62a692)

### System information

Windows 10, Vulkan, AMD Radeon RX 6700 XT

### Issue description

I created a tool script which dynamically creates controls in the _Ready function.
For one of the controls I need to set the AnchorPosition to CenterLeft.

However when setting AnchorPreset in the _Ready or _Init function the editor shows that the property has this value, but the control still remains in it's default position.

Doing the same using set_deffered works as expected.

### Steps to reproduce

Create a control and assign a tool script.

In this tool script try to set an AnchorPreset in _ready like FullRect or CenterLeft.

When you now instantiate the control in another script or add this node from the node selection (only works if you add a class_name to the tool script)

### Minimal reproduction project

Instantiate a new instance of BuggyControl.gd (with Add node)
Control should be full size but stays in the default one instead.

[BugReport1.zip](https://github.com/godotengine/godot/files/9742687/BugReport1.zip)

The same happens with
anchor_preset = Control.PRESET_FULL_RECT
set_anchors_preset(Control.PRESET_FULL_RECT, true)
set_anchors_preset(Control.PRESET_FULL_RECT, false)

**Comment by TheDuriel** [comment]
Setting the anchors will not have any visual effects until your resize the viewport.

You want to set the anchors and margins.

**Comment by dotlogix** [comment]
@TheDuriel same happens if I also set the offset, anchor positions, size and the preset.
I can also set the values during _init, so I would assume they should have a visual impact in the editor

Could you give an example what you propose me to do?

**Comment by ivandortulov** [comment]
I know this is an old issue and there have not been any updates but I would also like to know how to resolve it. 

In my scenario, I have a Control node that represents a popup. I want to be able to place the popup in different parts of the screen using code. 

I use the following code:

```gdscript
func show_popup(
	title: String,
	message: String,
	popup_position: Types.EPopupPosition = Types.EPopupPosition.Center
):
	self.popup.message = message

	match popup_position:
		Types.EPopupPosition.Center:
			self.popup.size.x = 3000
			self.popup.set_anchors_preset(Control.PRESET_CENTER, false)

		Types.EPopupPosition.Left:
			self.popup.size.x = 1500
			self.popup.set_anchors_preset(Control.PRESET_CENTER_LEFT, false)

		Types.EPopupPosition.Right:
			self.popup.size.x = 1500
			self.popup.set_anchors_preset(Control.PRESET_CENTER_RIGHT, false)

	self.popup.show()
```

However, when I run the game, my Control node is not centered properly. I tried to trigger an update by making the viewport emit the `size_changed` signal but that did not work.

```gdscript
get_viewport().size_changed.emit()
```

I changed my popup to use a PanelContainer as a root node and tried calling `queue_sort` on it, but that did not work either.

**Comment by ivandortulov** [comment]
> You didn't set the growth direction and the pivot point before resizing and positioning.

What do you mean?

[Panel Container](https://i.ibb.co/YykY4j7/panelcontainer.png)

These are my settings. There is no growth setting. In the code, I can see that by default horizontal and vertical growth are set to 1. The only thing that happens automatically is the size is updated to fit the label I have inside my container. The position of the Container itself does not get updated.

I had to resort to this code:

```gdscript
func show_popup(
	title: String,
	message: String,
	popup_position: Types.EPopupPosition = Types.EPopupPosition.Center
):
	self.popup.message = message
	self.popup.size.y = 500

	match popup_position:
		Types.EPopupPosition.Center:
			self.popup.size.x = 3000
			#self.popup.set_anchors_preset(Control.PRESET_CENTER)

		Types.EPopupPosition.Left:
			self.popup.size.x = 1500
			#self.popup.set_anchors_preset(Control.PRESET_CENTER_LEFT)

		Types.EPopupPosition.Right:
			self.popup.size.x = 1500
			#self.popup.set_anchors_preset(Control.PRESET_CENTER_RIGHT)

	self.popup.show()
	self.popup.resized.emit()
	await get_tree().process_frame

	var viewport_size = Vector2(6272, 3528)

	match popup_position:
		Types.EPopupPosition.Center:
			self.popup.global_position = \
				viewport_size / 2.0 - self.popup.size / 2.0
		Types.EPopupPosition.Right:
			self.popup.global_position = \
				Vector2(viewport_size.x, viewport_size.y / 2.0) \
					- Vector2(self.popup.size.x, self.popup.size.y / 2.0) \
					- Vector2(128.0, 0.0)
```

**Comment by ivandortulov** [comment]
I managed to force my Control to update with the following code:

```
func popup(popup_position: Types.EPopupPosition):
	self.position = Vector2.ZERO

	match popup_position:
		Types.EPopupPosition.Center:
			self.size = Vector2(3000.0, 0.0)
			self.anchors_preset = Control.PRESET_CENTER
		Types.EPopupPosition.Left:
			self.size = Vector2(1500.0, 0.0)
			self.anchors_preset = Control.PRESET_CENTER_LEFT
		Types.EPopupPosition.Right:
			self.size = Vector2(1500.0, 0.0)
			self.anchors_preset = Control.PRESET_CENTER_RIGHT
		Types.EPopupPosition.Top:
			self.size = Vector2(3000.0, 0.0)
			self.anchors_preset = Control.PRESET_CENTER_TOP

	show()
```

It works perfectly, except when changing the preset from PRESET_CENTER_RIGHT to PRESET_CENTER_TOP. Then the popup is not updated vertically. Hiding and showing it again does update it then.

**Comment by raulsntos** [comment]
- Related: https://github.com/godotengine/godot/issues/71062

The `anchor_preset` property is not meant to be used in scripting, it's supposed to be internal but it was exposed by mistake.

From https://github.com/godotengine/godot/issues/71062#issuecomment-1377067366:

> Both `anchors_preset` and `layout_mode` are not supposed to be used in scripting. They are marked as `PROPERTY_USAGE_INTERNAL`, and are only relevant for the editor. They exist as properties because that's the only way to have them in the inspector and in a specific order.

You may be interested in the [`Node.SetAnchorsPreset`](https://docs.godotengine.org/en/4.1/classes/class_control.html#class-control-method-set-anchors-preset) method.

**Comment by ivandortulov** [comment]
You are correct @raulsntos. By using the `set_anchor_*` methods, it did start to work as expected:

```
func popup(popup_position: Types.EPopupPosition):
	var viewport_size = Vector2(6272, 3528)

	match popup_position:
		Types.EPopupPosition.Center:
			set_size(Vector2(3000.0, 500.0))
			set_anchors_and_offsets_preset(Control.PRESET_CENTER, Control.PRESET_MODE_KEEP_WIDTH, 128.0)
		Types.EPopupPosition.Left:
			set_size(Vector2(1500.0, 500.0))
			set_anchors_and_offsets_preset(Control.PRESET_CENTER_LEFT, Control.PRESET_MODE_KEEP_WIDTH, 128.0)
		Types.EPopupPosition.Right:
			set_size(Vector2(1500.0, 500.0))
			set_anchors_and_offsets_preset(Control.PRESET_CENTER_RIGHT, Control.PRESET_MODE_KEEP_WIDTH, 128.0)
		Types.EPopupPosition.Top:
			set_size(Vector2(3000.0, 500.0))
			set_anchors_and_offsets_preset(Control.PRESET_CENTER_TOP, Control.PRESET_MODE_KEEP_WIDTH, 128.0)

	self.show()
```

**Comment by pc9098** [comment]
I tried to reproduce the BugReport1.zip in the 4.3 and I feel like something is missing... How do you see the Control node if it is invisible?

**Comment by pc9098** [comment]
To the original poster.
